test/e2e/app/app.go

```diff
@@ -1106,8 +1117,31 @@ func (app *Application) verifyExtensionTx(height int64, payload string) error {
 
 // If extension size was not specified, then parseVoteExtension attempts to parse
 // the given extension data into a positive integer.
+// It also checks the non replay protected extension: its height and its data.
 // Otherwise it is the size of the extension.
-func parseVoteExtension(cfg *Config, ext []byte) (int64, error) {
+func parseVoteExtensions(cfg *Config, expHeight int64, ext, nonRpExt []byte) (int64, error) {
+	parts := strings.Split(string(nonRpExt), "|")
+	if len(parts) != 2 {
+		return 0, fmt.Errorf("non replay protected vote extension must have 2 parts (%d)", len(parts))
+	}
+	height, err := strconv.ParseInt(parts[0], 10, 64)
+	if err != nil {
+		return 0, err
+	}
+	if height != expHeight {
+		return 0, fmt.Errorf("non replay protected vote extension contains incorrect height (%d!=%d), risk of replay attack",
+			expHeight,
+			height,
+		)
+	}
+	xExt := hex.EncodeToString(ext)
+	if parts[1] != xExt {
+		return 0, fmt.Errorf("non replay protected vote extension contains incorrect data (%s!=%s)",
+			xExt,
+			parts[1],
+		)
+	}
+
 	if cfg.VoteExtensionSize == 0 {
```

Comment:
Is this parameter not impacting the new vote extension type?

Reply:
indirectly, with no impact on the check done before (L1138) as this dependency is already reflected in the value of `ext`

Reply:
I just reviewed the whole logic of `cfg.VoteExtensionSize` and LGTM